File: src/main/java/fi/metatavu/metaform/server/rest/translate/AuditLogEntryTranslator.java
package fi.metatavu.metaform.server.rest.translate;


import fi.metatavu.metaform.api.spec.model.AuditLogEntryType;
import fi.metatavu.metaform.server.persistence.model.AuditLogEntry;

import javax.enterprise.context.ApplicationScoped;

/**
 * Translator for audit log entries
 *
 * @author Katja Danilova
 */
@ApplicationScoped
public class AuditLogEntryTranslator {

  /**
   * Translates into REST AuditLogEntry
   *
   * @param auditLogEntry JPA auditLogEntry
   * @return REST AuditLogEntry
   */
  public fi.metatavu.metaform.api.spec.model.AuditLogEntry translateAuditLogEntry(AuditLogEntry auditLogEntry) {
    if (auditLogEntry == null) {
      return null;
    }

    fi.metatavu.metaform.api.spec.model.AuditLogEntry result = new fi.metatavu.metaform.api.spec.model.AuditLogEntry();
    result.setCreatedAt(auditLogEntry.getCreatedAt());
    result.setId(auditLogEntry.getId());
    result.setMessage(auditLogEntry.getMessage());
    result.setUserId(auditLogEntry.getUserId());
    result.setReplyId(auditLogEntry.getReplyId());
    AuditLogEntryType originalLogEntryType = auditLogEntry.getLogEntryType();
    result.setLogEntryType(AuditLogEntryType.fromValue(originalLogEntryType.toString()));
    return result;
  }
}

File: src/main/kotlin/fi/metatavu/metaform/server/rest/translate/AuditLogEntryTranslator.kt
package fi.metatavu.metaform.server.rest.translate

import fi.metatavu.metaform.server.persistence.model.AuditLogEntry
import jakarta.enterprise.context.ApplicationScoped

/**
 * Translator for audit log entries
 *
 * @author Katja Danilova
 */
@ApplicationScoped
class AuditLogEntryTranslator {
  /**
   * Translates into REST AuditLogEntry
   *
   * @param auditLogEntry JPA auditLogEntry
   * @return REST AuditLogEntry
   */
  fun translate(auditLogEntry: AuditLogEntry): fi.metatavu.metaform.api.spec.model.AuditLogEntry {
    return fi.metatavu.metaform.api.spec.model.AuditLogEntry(
      id = auditLogEntry.id,
      message = auditLogEntry.message,
      userId = auditLogEntry.userId,
      replyId = auditLogEntry.replyId,
      logEntryType = auditLogEntry.logEntryType,
      createdAt = auditLogEntry.createdAt
    )
  }
}